examples/run_lm_finetuning.py

```diff
@@ -82,11 +83,11 @@
 
 
 class TextDataset(Dataset):
-    def __init__(self, tokenizer, args, file_path="train", block_size=512):
+    def __init__(self, tokenizer: PreTrainedTokenizer, args, file_path="train", block_size=512):
         assert os.path.isfile(file_path)
         directory, filename = os.path.split(file_path)
         cached_features_file = os.path.join(
-            directory, args.model_name_or_path + "_cached_lm_" + str(block_size) + "_" + filename
+            directory, args.model_type + "_cached_lm_" + str(block_size) + "_" + filename
```

Comment:
This will break backward compatibility with users that have already tokenized their text into a file (not too bad), and won't allow multiple cache files for different checkpoints of the same model (annoying).

Couldn't we use the `model_name_or_path` and fallback to the `model_type` if training from scratch?

Reply:
I guess we could, but honestly I don't even know if the previous behavior was very well defined, for instance if `model_name_or_path` is a path (like folder/folder/path)

For backward compatibility, I'm not sure a lot of users of an example script would upgrade the example script anyways (in the middle of using it/training models), what do you think?